lib/shared/sdk/index.js

```diff
@@ -35,44 +38,223 @@ const ADAPTERS = [
  * @type {Object<String,Adapter>}
  * @constant
  */
-sdk.adapters = _.reduce(ADAPTERS, (adapters, Adapter) => {
-  adapters[Adapter.name] = new Adapter()
+SDK.adapters = _.reduce(ADAPTERS, (adapters, Adapter) => {
+  adapters[Adapter.id] = new Adapter()
   return adapters
 }, {})
 
+/* eslint-disable lodash/prefer-lodash-method */
+
 /**
- * @summary Scan for drives using all registered adapters
- * @function
- * @public
- *
- * @description
- * The options object contains options for all the registered
- * adapters. For the `standard` adapter, for example, place
- * options in `options.standard`.
- *
- * @param {Object} options - options
- * @fulfil {Object[]} - drives
- * @returns {Promise}
- *
+ * Adapter Scanner
+ * @class Scanner
+ */
+SDK.Scanner = class Scanner extends EventEmitter {
+  /**
+   * @summary Adapter Scanner constructor
+   * @param {Object<String,Object>} [options] - device adapter options
+   * @param {Object} [options.adapters] - map of external device adapters
+   * @example
+   * new SDK.Scanner({
+   *   standard: { ... },
+   *   usbboot: { ... }
+   * })
+   */
+  constructor (options = {}) {
+    // Inherit from EventEmitter
+    super()
+
+    this.options = options
+    this.isScanning = false
+    this.adapters = new Map()
+
+    // Bind event handlers to own context to facilitate
+    // removing listeners by reference
+    this.onDevices = this.onDevices.bind(this)
+    this.onError = this.onError.bind(this)
+
+    this.init()
+  }
+
+  /**
+   * @summary Initialize adapters
+   * @private
+   * @example
+   * // Only to be used internally
+   * this.init()
+   */
+  init () {
+    debug('scanner:init', this)
+    _.map(_.keys(this.options), (adapterId) => {
```

Comment:
Should this be `_.each`, given we're not returning from the callback, nor assigning the results?

Reply:
Fixed!